src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -902,3 +915,31 @@ def _rehome(self, l):
       return HermeticDistribution('.jdk', local_distribution)
     else:
       return local_distribution
+
+  def  _collect_invalid_compile_dependencies(self, compile_target, invalid_target_set):
+    # Collects all invalid dependencies that are not dependencies of other invalid dependencies
```

Comment:
docstring?

Reply:
xx'd the whole thing in favor of a template method.